Pod/Classes/Heartbeat/SEGAdobeHeartbeat.m

```diff
@@ -5,4 +5,58 @@
 //  Created by ladan nasserian on 12/2/17.
 //
 
-#import <Foundation/Foundation.h>
+#import "SEGAdobeHeartbeat.h"
+#import <Analytics/SEGAnalytics.h>
+
+@interface VideoAnalyticsProvider : NSObject <ADBMediaHeartbeatDelegate>
+
+id<ADBMediaHeartbeatDelegate> playBackDelegate();
```

Comment:
Is this supposed to be an instance method? Why is it as a C style function?

Reply:
Yeah I guess my thought process here is that I would trigger the Video Heartbeat logic from `track`in SEGAdobeAnalytics.m , and having all the functions self contained in this file including the initialization of the ADBMobileHeartbeat, and just passing that into each function.

Could be a wrong approach here.

So in SEGAdobeAnalytics.m
```
- (void)track:(SEGTrackPayload *)payload
if event is equal to adobe video events
        trackHeartbeatEvents(payload, self.settings);
```

then in this file 

``` 
if video playback event..
void videoPlaybackStarted(SEGTrackPayload *payload, NSDictionary *settings)
{
    ADBMediaHeartbeatConfig *config = createConfig(payload, settings);
    
    ADBMediaHeartbeat *ADBMediaHeartbeat = initializeMediaHeartbeat(config, playBackDelegate());
    [ADBMediaHeartbeat trackPlay];

}

Reply:
I think the thinking behind the approach is fine - the implementation is not though. There's a lint warning for this (run make lint) (should show up in Xcode too), that should guide you in the right direction to fix this.